api/api/clubs/tests.py

```diff
@@ -55,3 +58,98 @@ def testListClubs_hundredClubExistsInDb_returnsFiftyClubInFirstResponse(self):
+
+        test_club = Club.objects.create(name="Bitbyte - The Programming Club",
+                                        category="S&T",
+                                        description="Some desc here",
+                                        email="theprogclub@iiitdmj.ac.in",
+                                        logo="#")
+        # when
+        response = self.client.put(f'/clubs/{test_club.id}/', {'name': 'Bitbyte',
+                                                               'category': "S&T",
+                                                               'description': "Some desc here",
+                                                               'email': "theprogclub@iiitdmj.ac.in",
+                                                               'logo': "https://iiitdmj.ac.in"})
+
+        # then
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    @pytest.mark.django_db
+    def testUpdateClubs_positionHolderRequest_updatesSuccessful(self):
+        # given
+        user = User.objects.create(
+            email='test@user.com', first_name='Test', last_name='User')
+        self.client.force_authenticate(user)
+
+        test_club = Club.objects.create(name="Bitbyte - The Programming Club",
+                                        category="S&T",
+                                        description="Some desc here",
+                                        email="theprogclub@iiitdmj.ac.in",
+                                        logo="#")
+
+        Roles.objects.create(name='Coordinator',
+                             club=test_club,
+                             user=user,
+                             assigned_at=date.today())
+        # when
+        response = self.client.put(f'/clubs/{test_club.id}/', {'name': 'changed_club_name',
+                                                               'category': "S&T",
+                                                               'description': "Some desc here",
+                                                               'email': "theprogclub@iiitdmj.ac.in",
+                                                               'logo': "https://iiitdmj.ac.in"})
+
+        # then
+        updated_club = Club.objects.get(pk=test_club.id)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, {'id': 1,
+                                         'name': 'changed_club_name',
+                                         'category': 'S&T',
+                                         'logo': 'https://iiitdmj.ac.in',
+                                         'description': 'Some desc here',
+                                         'email': 'theprogclub@iiitdmj.ac.in'})
+        self.assertEqual(updated_club.name, 'changed_club_name')
+        self.assertEqual(updated_club.category, 'S&T')
+
+    @pytest.mark.django_db
+    def testUpdateClubs_incompleteInformationInRequest_throwsBadRequest(self):
```

Comment:
From the name it's not clear what information is not provided. Maybe a document it at `when`